### PR TITLE
Vahagn: task from 151 to 160

### PR DESCRIPTION
**Pull request by hsp-vahagn**
Vahagn: task from 151 to 160

### CODE REVIEW

File: src/main/java/am/hsp/apps/seua/Task_155.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+		System.out.println(getDivisor());
+	}
+
+	/**
+	 * get divisor
+	 */
+	private int getDivisor() {
+		int sum = 0;
+		for (int n = 10; n <= 99; n++) {
```
[the remark this answers is not in the document]
Comment (hsp-mona) [reply src/main/java/am/hsp/apps/seua/Task_155.java]: To have a generic method you need to pass dynamic data as its parameters.The method name could be simplified also:
getSumOfMultiples(int from, int to, int divisor)

File: src/main/java/am/hsp/apps/seua/Task_156.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	public void solve() {
+		System.out.println(getDivisor());
+	}
+
+	/**
+	 * get divisor
+	 */
+	private long getDivisor() {
+		long product = 1;
+		for (int n = 10; n <= 99; n++) {
```
Comment (hsp-mona) [inline src/main/java/am/hsp/apps/seua/Task_156.java]: the same issue as in previous task...

File: src/main/java/am/hsp/apps/seua/Task_157.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+		System.out.println(getSumOfNonMultipleOfGivenNumber());
+	}
+
+	/**
+	 * get the sum of three-digit numbers which no multiple of 5
+	 */
+	private int getSumOfNonMultipleOfGivenNumber() {
+		int sum = 0;
+		for (int number = 100; number <= 999; number++) {
+			if (!isDivisor(number, DIVISOR_VALUE)) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_157.java]: As we discussed in one of our previous lessons, Your code will be more cleaner and understandable if you will have a separate method for checking a negative conditions:
isNotDivisor

File: src/main/java/am/hsp/apps/seua/Task_157.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+	public void solve() {
+		System.out.println(getSumOfNonMultipleOfGivenNumber());
+	}
+
+	/**
+	 * get the sum of three-digit numbers which no multiple of 5
+	 */
+	private int getSumOfNonMultipleOfGivenNumber() {
+		int sum = 0;
+		for (int number = 100; number <= 999; number++) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_157.java]: Remove magic numbers

File: src/main/java/am/hsp/apps/seua/Task_151.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+		System.out.println(getDivisor(number));
+
+	}
+
+	/**
+	 * get divisor
+	 */
+	private int getDivisor(int n) {
+		int sum = 0;
+		for (int i = 1; i <= n; i++) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_151.java]: I guess there is no need to continue the loop until divisor will be equal to number. The max divisor should be "number / 2".
If you are doing this to include also the "number" as its divisor, you just need to add the number to "sum" before the loop...
**MAKE THIS FIX IN ALL TASKS THAT YOU HAVE SIMILAR FUNCTIONALITY**

File: src/main/java/am/hsp/apps/seua/Task_153.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+		System.out.println(getDivisor(number));
+	}
+
+	/**
+	 * get divisor
+	 */
+	private int getDivisor(int number) {
+		int sum = 0;
+		for (int i = 1; i < number; i++) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_153.java]: You can decrease loop iterations, maximum divisor could be "**number - balance**"

File: src/main/java/am/hsp/apps/seua/Task_154.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+	public void solve() {
+		int number = getUserInput("Please enter the number");
+
+		System.out.println(getDivisor(number));
+	}
+
+	/**
+	 * get divisor
+	 */
+	private int getDivisor(int number) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_154.java]: see comments in previous task...

File: src/main/java/am/hsp/apps/seua/Task_155.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+	@Override
+	public void solve() {
+
+		System.out.println(getDivisor());
+	}
+
+	/**
+	 * get divisor
+	 */
+	private int getDivisor() {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_155.java]: see comments in previous tasks...

File: src/main/java/am/hsp/apps/seua/Task_160.java [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+	public void solve() {
+		System.out.println(isSquareOfNaturalNumber());
+	}
+
+	/**
+	 * is given number square of natural number
+	 */
+	private boolean isGivenNumberSquareOfNaturalNumber(int a, int i) {
+		for (i = 1; i < a; i++) {
+			if (a == i * i) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_160.java]: use pow method to not violate dry principle...

File: src/main/java/am/hsp/apps/seua/base/TaskBase.java [hunk 10]
```diff
@@ -115,4 +115,18 @@ protected boolean isArithmeticProgression(int number1, int number2, int number3)
 	protected boolean isGeometricProgression(int number1, int number2, int number3) {
 		return (double) number2 / number1 == (double) number3 / number2;
 	}
+
+	/**
+	 * is divisor
+	 */
+	protected boolean isDivisor(int number1, int number2) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/base/TaskBase.java]: Bad param names. I guess one of params is considered as divisor. So name it accordingly

File: src/main/java/am/hsp/apps/seua/Task_154.java [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+	public void solve() {
+		int number = getUserInput("Please enter the number");
+
+		System.out.println(getSumOfDivisorsByGivenBalance(number));
+	}
+
+	/**
+	 * get sum of divisors by given balancer
+	 */
+	private int getSumOfDivisorsByGivenBalance(int number) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_154.java]: Product not Sum

**Pull request merged**